npa.js

```diff
@@ -18,6 +18,17 @@ const isFilename = /[.](?:tgz|tar.gz|tar)$/i
 function npa (arg, where) {
   let name
   let spec
+  if (arg && arg.toString().match(/^\[object/)) {
+    // we got an object but it's not a proper npa instance
+    if (arg.name && arg.rawSpec) {
+      return npa.resolve(arg.name, arg.rawSpec, where)
+    } else {
+      return npa(arg.raw, arg.where || where)
```

Comment:
I'm pretty sure this should be `where || arg.where` because if we are passing in a where then it should override wherever the one in this spec came from (because that may have been on a different machine or a directory that was `mv`ed or who knows what).